contracts-periphery/src/WithdrawHook.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+pragma abicoder v2;
+
+import "openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+import "src/utils/ISwapRouter.sol";
+
+contract WithdrawHook {
+
+  ISwapRouter internal immutable swapRouter;
+  address internal constant WETH9 = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+
+  constructor(ISwapRouter _swapRouter) {
+  swapRouter = _swapRouter;
+  }
+
+  function tokensWithdrawn(
+    uint256 _amount,
+    address _stealthAddr,
+    address _acceptor,
+    address _tokenAddr,
+    address _sponsor,
+    uint256 _sponsorFee,
+    bytes memory _data
+  ) external{
+    // Plz recommend checks that should happen here
+    _swapForEth(_amount, _stealthAddr, _acceptor, _tokenAddr, _sponsor, _sponsorFee, _data);
+  }
+
+  function _swapForEth(
+    uint256 _amount,
+    address _stealthAddr,
+    address _acceptor,
+    address _tokenAddr,
+    address _sponsor,
+    uint256 _sponsorFee,
+    bytes memory _data) internal {
+    uint256 senderBalance = _amount;
+    if(_sponsorFee > 0) {
```

Comment:
Not sure what this section of the code is supposed to be doing. The withdrawer [will have paid](https://github.com/ScopeLift/umbra-protocol/blob/master/contracts-core/contracts/Umbra.sol#L271-L273) the sponsor fee already as part of the withdrawal. These params are simply passed forward to the hook in case the hook needed them for any reason. In this case, they can probably be ignored. Unless I'm missing something with the intent here?